libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts

```diff
@@ -18,6 +25,58 @@ export const organizations = createQueryKeys('organizations', {
       return response.data
     },
   }),
+  gitTokens: (organizationdId) => ({
+    queryKey: [organizationdId],
+    async queryFn() {
+      const response = await organizationApi.listOrganizationGitTokens(organizationdId)
+      return response.data.results
+    },
+  }),
+  authProviders: (organizationId) => ({
+    queryKey: [organizationId],
+    async queryFn() {
+      const response = await gitApi.getOrganizationGitProviderAccount(organizationId)
+      return response.data
+    },
+  }),
+  repositories: (organizationId, gitProvider, gitToken) => ({
+    queryKey: [organizationId, gitProvider, gitToken],
+    async queryFn() {
+      if (gitProvider === GitProviderEnum.GITHUB) {
```

Comment:
If you type `gitProvider` as `gitProvider: keyof typeof GitProviderEnum` you can use ts-pattern `match` and be exhaustive.